preprocessing/shift_generation.py

```diff
@@ -190,42 +193,80 @@ def get_shifts_covered_by_off_shifts(shifts, off_shifts):
 
 def get_shifts_violating_daily_rest(root, staff, shifts_per_day):
     """
-    Returns a dict, with "employee" as key and another dict as value. The new dict uses "shift"
-    as key and have a list of shifts that violates daily rest for "employee" if "shift" is worked.
+    Returns:
+        * violating_shift_sequences:    A dict, with "employee" as key and another dict as value. The new dict
+                                        uses "shift" as key, with value: a list of shifts sequences that, if two or
+                                        more of them are worked, contributes to violating daily rest for "employee"
+                                        if "shift" is worked.
+        * violating_shift_combinations: A dict, with "employee" as key and another dict as value. The new dict uses
+                                        "shift" as key, with value: a list of shifts that violates daily rest for
+                                        "employee" if "shift" is worked.
     """
 
     employees = staff["employees"]
     daily_rest = staff["employee_daily_rest"]
     daily_offset = staff["employee_daily_offset"]
-    violating_shifts = tupledict()
+    violating_shift_sequences = tupledict()
+    violating_shift_combinations = tupledict()
 
     for e in employees:
-        violating_shifts[e] = tupledict()
+        violating_shift_sequences[e] = tupledict()
+        violating_shift_combinations[e] = tupledict()
         for day in shifts_per_day:
+            # Check if daily rest is fulfilled naturally
             if not (already_daily_off_shift(root, daily_offset[e], daily_rest[e], day)):
                 for shift in shifts_per_day[day]:
                     if day != 0:
+                        # Checking shifts the day before
                         for s in shifts_per_day[day - 1]:
+                            # Checking if the shift from the day before is ending in employee specific day
                             if s[0] + s[1] > (24 * int(day)) + daily_offset[e]:
-                                if shift[0] - (s[0] + s[1]) < daily_rest[e] and shift[0] >= (
-                                    s[0] + s[1]
-                                ):
+                                # Checking if daily rest occurs between the shifts and that they are not overlapping
+                                if shift[0] - (s[0] + s[1]) < daily_rest[e] and shift[0] >= (s[0] + s[1]):
```

Comment:
Is my understanding correct in that this if-statement should check that
1. There is _not_ enough time for sufficient rest between the shifts

and

2. `shift` and `s` should not overlap

Reply:
Yes!